crates/bevy_ui/src/render/mod.rs

```diff
@@ -435,11 +435,14 @@ pub fn prepare_uinodes(
 
         let transformed_rect_size = extracted_uinode.transform.transform_vector3(rect_size);
 
-        // Cull nodes that are completely clipped
-        if positions_diff[0].x - positions_diff[1].x >= transformed_rect_size.x
-            || positions_diff[1].y - positions_diff[2].y >= transformed_rect_size.y
-        {
-            continue;
+        // Don't try to cull nodes that have a rotation
+        if extracted_uinode.transform.x_axis[1] == 0.0 {
```

Comment:
I'm not super aware of the UI rendering stuff but a quick glance suggests it generates quads similar to how it does for sprites and then this is just trying to cull stuff that is offscreen.

However, I would have thought that rotations would be around the z axis. I don't understand the mathematical reasoning behind looking at the y component of the x axis of the transform to identify if the node has been rotated. From what I recall, a rotation matrix about the z axis would have some combination of cos(angle) and sin(angle) in the x and y components of the x and y axes (i.e. the upper-left 2x2 part.) In that case a rotation about the z axis would likely but not necessarily have non-zero .x.y component given that the sine or cosine of the angle could be 0 but then the other would be 1 and so that would still be a rotation.

Reply:
See: https://github.com/bitshifter/glam-rs/blob/main/src/f32/scalar/mat4.rs#L409

Reply:
For this value to be 0, either we have a rotation of 0 or a rotation of π. In both those case, not culling will happen correctly as the x and y coordinates of corners have valid values according to the culling check. For all other angles than those two, the check  is bypassed.
It's dirty but I didn't want to put more calculations here.

Reply:
Great! Could you add a comment that clarifies that in the code?